### PR TITLE
feat/server tests

### PR DESCRIPTION
**Pull request by jenr24G**
adds tests to the ApiContext and axum Router

### CODE REVIEW

File: server/src/queries/mod.rs [hunk 1]
```diff
@@ -1,29 +1,78 @@
...
+        network: &QueryRequestFilter,
+    ) -> Result<i64, sqlx::Error> {
+        let row: (i64,) = sqlx::query_as("SELECT MAX(height) FROM blocks;")
+            .fetch_one(match network {
+                QueryRequestFilter::Mainnet => &self.mainnet_db,
+                _ => &self.devnet_db,
+            })
+            .await?;
+        Ok(row.0)
+    }
```
Comment (mxnkarou) [inline server/src/queries/mod.rs:61]: This function should not be part of the APIContext - it's a query & not a direct method on the context itself.

File: server/src/queries/mod.rs [hunk 2]
```diff
@@ -1,29 +1,78 @@
...
+
+    /// By default we get all signals every query.
+    /// This is wildly inefficient & we should work towards an alternative.
+    /// Since we're essentially only interested in memo's - maybe a way to decode base58 on the DB side?
+    pub async fn get_signals(
```
Comment (mxnkarou) [inline server/src/queries/mod.rs:66]: This function should not be part of the APIContext - it's a query & not a direct method on the context itself.

File: server/src/queries/mod.rs [hunk 3]
```diff
@@ -39,5 +88,49 @@ pub async fn get_signals(db: &Pool<Postgres>) -> Result<Vec<DBResponse>, sqlx::E
...
+    }
+}
+
+pub async fn build_router(context: ApiContext) -> Router {
+    let cors = CorsLayer::new()
+        .allow_methods([Method::GET, Method::POST])
+        .allow_origin(Any);
+    axum::Router::build_v1(&context.config)
+        .layer(ServiceBuilder::new().layer(cors).layer(Extension(context)))
+}
```
Comment (mxnkarou) [inline server/src/queries/mod.rs:101]: This function should be defined in `router/mod.rs` - it's a bit confusing having this in the queries module.

File: server/src/queries/mod.rs [hunk 4]
```diff
@@ -39,5 +88,49 @@ pub async fn get_signals(db: &Pool<Postgres>) -> Result<Vec<DBResponse>, sqlx::E
...
+
+pub fn create_config() -> Option<Config> {
```
Comment (mxnkarou) [inline server/src/queries/mod.rs:103]: As far as I can tell this function is solely used in a testing context - so it should find it's definition there as well.
Most likely - inside ` __mocks__` & renamed to `create_mock_config`

File: server/src/router/api/keyword.rs [hunk 5]
```diff
@@ -7,58 +7,58 @@ use axum::{
...
+        axum::Json("Error: Network param not provided."),
     )
-    .into_response()
-}
\ No newline at end of file
+        .into_response()
+}
+
+#[cfg(test)]
+mod tests {}
```
Comment (mxnkarou) [inline server/src/router/api/keyword.rs:64]: This unused testing module can be removed.

**Review by mxnkarou — CHANGES_REQUESTED** [review]
Left a few comments!

File: server/src/queries/mod.rs (hunk 1 above)
Comment (jenr24G) [reply server/src/queries/mod.rs]: figured queries would be taking place from the context of the API

Comment (jenr24G) [reply server/src/queries/mod.rs]: brb changing 30 method receivers into regular function calls

**Comment by mxnkarou** [comment]
Deprecated.

**Pull request closed without merging**